test/forward_merger.test.ts

```diff
@@ -0,0 +1,300 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ForwardMerger } from "../src/plugins/ForwardMerger/forward_merger";
+import { makePushContext } from "./fixtures/contexts/push";
+import { mockConfigGet, mockContextRepo, mockCreatePR, mockListBranches, mockMerge, mockPaginate } from "./mocks";
+import { default as repoResp } from "./fixtures/responses/context_repo.json";
+import { makeConfigReponse } from "./fixtures/responses/get_config";
+
+describe("Forward Merger", () => {
+  beforeEach(() => {
+    mockCreatePR.mockReset();
+    mockMerge.mockReset();
+    mockListBranches.mockReset();
+    mockPaginate.mockReset();
+  });
+
+  beforeAll(() => {
+    mockContextRepo.mockReset();
+    mockContextRepo.mockReturnValue(repoResp);
+    mockConfigGet.mockResolvedValue(
+      makeConfigReponse({
+        forward_merger: true,
+      })
+    );
+  });
+
+  test("mergeForward should not run on push to non-versioned branch", async () => {
+    const context = makePushContext({
+      ref: "refs/heads/unversioned",
+    });
+    const forwardMerger = new ForwardMerger(context, context.payload);
+    const mockGetBranches = jest.fn().mockName("getBranches");
+    forwardMerger.getBranches = mockGetBranches;
+    await forwardMerger.mergeForward();
+
+    expect(mockGetBranches).not.toBeCalled();
+  });
+
+  test("mergeForward should not run when plugin is disabled", async () => {
+    const context = makePushContext({
+      ref: "refs/heads/branch-21.12",
+    });
+    const forwardMerger = new ForwardMerger(context, context.payload);
+    const mockGetBranches = jest.fn().mockName("getBranches");
+    forwardMerger.getBranches = mockGetBranches;
+    const mockPluginIsDisabled = jest
+      .fn()
+      .mockName("pluginIsDisabled")
+      .mockResolvedValue(true);
+    forwardMerger.pluginIsDisabled = mockPluginIsDisabled;
+    await forwardMerger.mergeForward();
+
+    expect(mockPluginIsDisabled).toBeCalled();
+    expect(mockGetBranches).not.toBeCalled();
+  });
+
+
+  test("mergeForward should not open PR on invalid next branch", async () => {
+    const context = makePushContext({
+      ref: "refs/heads/branch-22.02",
+    });
+    const forwardMerger = new ForwardMerger(context, context.payload);
+    forwardMerger.getBranches = jest
+      .fn()
+      .mockName("getBranches")
+      .mockResolvedValue(null);
+    forwardMerger.sortBranches = jest
+      .fn()
+      .mockName("sortBranches")
+      .mockReturnValue(null);
+    const nextBranch = null;
+    const mockGetNextBranch = jest
+      .fn()
+      .mockName("getNextBranch")
+      .mockReturnValue(nextBranch);
+    forwardMerger.getNextBranch = mockGetNextBranch;
+    mockCreatePR.mockResolvedValue(null);
+
+    await forwardMerger.mergeForward();
+
+    expect(mockCreatePR).not.toBeCalled();
+  });
+
+
+  test("should comment failure on PR if merge is successful", async () => {
```

Comment:
```suggestion
  test("should comment success on PR if merge is successful", async () => {
```

I assume this is a typo